nucleus/core/kernel/src/main/java/com/sun/enterprise/v3/server/AppServerStartup.java

```diff
@@ -222,18 +220,10 @@ public synchronized void start() {
             // See issue #5596 to know why we set context CL as common CL.
             Thread.currentThread().setContextClassLoader(
                     commonCLS.getCommonClassLoader());
-            if (state.compareAndSet(State.INITIAL, State.STARTING) || state.compareAndSet(State.SHUT_DOWN, State.STARTING)) {
```

Comment:
This seems like a stronger way of checking the startup state of the server - why has this been removed?

Reply:
I messed up my commit history and mistakenly removed the previous commit made by Patrick. I will fix it and update the PR. Thanks for pointing out